### PR TITLE
[Enhancement] keep tablet scheduling related config naming consistent

### LINKED ISSUE #8871: Keep clone related configuration names consistent
## Enhancement
Currently the naming style can be very hard to understand by our users.
For compatibility reason, old name is still supported, we use an alias annotation mechnism to achieve this.

### PR DESCRIPTION
**Pull request by nshangyiming**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8871 

## Problem Summary(Required) ：

Add a new alias annotation method to keep forward compatibility in ConfigBase class.
Rename and rearrange some configurations in Config.java.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 62 / 104 (59.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|1|00.00%|[154]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|3|00.00%|[625, 628, 631]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|9|00.00%|[108, 109, 111, 112, 114, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/http/rest/SetConfigAction.java|0|11|00.00%|[68, 69, 70, 71, 73, 74, 75, 76, 77, 79, 94]|
|:large_blue_circle:|com/starrocks/http/rest/RowCountAction.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/http/rest/GetDdlStmtAction.java|0|1|00.00%|[104]|
|:large_blue_circle:|com/starrocks/http/rest/MetaReplayerCheckAction.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|1|6|16.67%|[655, 669, 715, 852, 853]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|4|10|40.00%|[174, 283, 284, 357, 361, 486]|
|:large_blue_circle:|com/starrocks/clone/DiskAndTabletLoadReBalancer.java|7|9|77.78%|[249, 1565]|
|:large_blue_circle:|com/starrocks/common/ConfigBase.java|32|34|94.12%|[91, 265]|
|:large_blue_circle:|com/starrocks/clone/TabletChecker.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/clone/ColocateTableBalancer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 61 / 104 (58.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|1|00.00%|[154]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|3|00.00%|[625, 628, 631]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|9|00.00%|[108, 109, 111, 112, 114, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/http/rest/SetConfigAction.java|0|11|00.00%|[68, 69, 70, 71, 73, 74, 75, 76, 77, 79, 94]|
|:large_blue_circle:|com/starrocks/http/rest/RowCountAction.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/http/rest/GetDdlStmtAction.java|0|1|00.00%|[104]|
|:large_blue_circle:|com/starrocks/http/rest/MetaReplayerCheckAction.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|1|6|16.67%|[655, 669, 715, 852, 853]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|4|10|40.00%|[174, 283, 284, 357, 361, 486]|
|:large_blue_circle:|com/starrocks/clone/DiskAndTabletLoadReBalancer.java|7|9|77.78%|[249, 1565]|
|:large_blue_circle:|com/starrocks/common/ConfigBase.java|31|34|91.18%|[87, 92, 266]|
|:large_blue_circle:|com/starrocks/clone/TabletChecker.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/clone/ColocateTableBalancer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 64 / 109 (58.72%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|1|00.00%|[154]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|3|00.00%|[625, 628, 631]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|9|00.00%|[108, 109, 111, 112, 114, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/http/rest/SetConfigAction.java|0|11|00.00%|[68, 69, 70, 71, 73, 74, 75, 76, 77, 79, 94]|
|:large_blue_circle:|com/starrocks/http/rest/RowCountAction.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/http/rest/GetDdlStmtAction.java|0|1|00.00%|[104]|
|:large_blue_circle:|com/starrocks/http/rest/MetaReplayerCheckAction.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|1|6|16.67%|[655, 669, 715, 852, 853]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|4|10|40.00%|[174, 283, 284, 357, 361, 486]|
|:large_blue_circle:|com/starrocks/clone/DiskAndTabletLoadReBalancer.java|7|9|77.78%|[249, 1565]|
|:large_blue_circle:|com/starrocks/common/ConfigBase.java|34|39|87.18%|[72, 73, 95, 100, 274]|
|:large_blue_circle:|com/starrocks/clone/TabletChecker.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/clone/ColocateTableBalancer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|3|3|100.00%|[]|

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 65 / 110 (59.09%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|1|00.00%|[154]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|3|00.00%|[625, 628, 631]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|9|00.00%|[108, 109, 111, 112, 114, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/http/rest/SetConfigAction.java|0|11|00.00%|[68, 69, 70, 71, 73, 74, 75, 76, 77, 79, 94]|
|:large_blue_circle:|com/starrocks/http/rest/RowCountAction.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/http/rest/GetDdlStmtAction.java|0|1|00.00%|[104]|
|:large_blue_circle:|com/starrocks/http/rest/MetaReplayerCheckAction.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|1|6|16.67%|[655, 669, 715, 852, 853]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|4|10|40.00%|[174, 283, 284, 357, 361, 486]|
|:large_blue_circle:|com/starrocks/clone/DiskAndTabletLoadReBalancer.java|7|9|77.78%|[249, 1565]|
|:large_blue_circle:|com/starrocks/common/ConfigBase.java|35|40|87.50%|[72, 73, 97, 102, 276]|
|:large_blue_circle:|com/starrocks/clone/TabletChecker.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/clone/ColocateTableBalancer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|3|3|100.00%|[]|

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 65 / 110 (59.09%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|1|00.00%|[154]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|3|00.00%|[625, 628, 631]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|9|00.00%|[108, 109, 111, 112, 114, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/http/rest/SetConfigAction.java|0|11|00.00%|[68, 69, 70, 71, 73, 74, 75, 76, 77, 79, 94]|
|:large_blue_circle:|com/starrocks/http/rest/RowCountAction.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/http/rest/GetDdlStmtAction.java|0|1|00.00%|[104]|
|:large_blue_circle:|com/starrocks/http/rest/MetaReplayerCheckAction.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|1|6|16.67%|[655, 669, 715, 852, 853]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|4|10|40.00%|[174, 283, 284, 357, 361, 486]|
|:large_blue_circle:|com/starrocks/clone/DiskAndTabletLoadReBalancer.java|7|9|77.78%|[249, 1565]|
|:large_blue_circle:|com/starrocks/common/ConfigBase.java|35|40|87.50%|[72, 73, 97, 102, 276]|
|:large_blue_circle:|com/starrocks/clone/TabletChecker.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/clone/ColocateTableBalancer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|3|3|100.00%|[]|

**Comment by nshangyiming** [comment]
the code smell can be ignored for historical reasons

File: fe/fe-core/src/test/resources/conf/config_test.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+tablet_sched_slot_num_per_path = 3
```
Comment (chaoyli) [inline fe/fe-core/src/test/resources/conf/config_test.properties:1]: what's the purpose of the three properites?

File: fe/fe-core/src/test/resources/conf/config_test3.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+schedule_slot_num_per_path_only_for_test = 5
```
Comment (chaoyli) [inline fe/fe-core/src/test/resources/conf/config_test3.properties:1]: This test alias can be removed in production code paths.

File: fe/fe-core/src/test/resources/conf/config_test.properties (hunk 1 above)
Comment (nshangyiming) [reply fe/fe-core/src/test/resources/conf/config_test.properties:1]: They are loaded in the ut to test some behaviors of ConfigBase class.

File: fe/fe-core/src/test/resources/conf/config_test3.properties (hunk 2 above)
Comment (nshangyiming) [reply fe/fe-core/src/test/resources/conf/config_test3.properties:1]: i will fix this

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 68 / 114 (59.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/MigrationAction.java|0|1|00.00%|[154]|
|:large_blue_circle:|com/starrocks/catalog/LocalTablet.java|0|3|00.00%|[625, 628, 631]|
|:large_blue_circle:|com/starrocks/http/rest/RestBaseAction.java|0|9|00.00%|[108, 109, 111, 112, 114, 117, 118, 119, 120]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|0|1|00.00%|[1899]|
|:large_blue_circle:|com/starrocks/http/rest/SetConfigAction.java|0|11|00.00%|[68, 69, 70, 71, 73, 74, 75, 76, 77, 79, 94]|
|:large_blue_circle:|com/starrocks/http/rest/RowCountAction.java|0|1|00.00%|[109]|
|:large_blue_circle:|com/starrocks/http/rest/GetDdlStmtAction.java|0|1|00.00%|[104]|
|:large_blue_circle:|com/starrocks/http/rest/MetaReplayerCheckAction.java|0|1|00.00%|[58]|
|:large_blue_circle:|com/starrocks/clone/TabletSchedCtx.java|1|6|16.67%|[655, 669, 715, 852, 853]|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|4|10|40.00%|[174, 283, 284, 357, 361, 486]|
|:large_blue_circle:|com/starrocks/clone/DiskAndTabletLoadReBalancer.java|7|9|77.78%|[249, 1565]|
|:large_blue_circle:|com/starrocks/common/ConfigBase.java|35|40|87.50%|[72, 73, 97, 102, 276]|
|:large_blue_circle:|com/starrocks/clone/TabletChecker.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/clone/ColocateTableBalancer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/SingleRangePartitionDesc.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowConfigStmt.java|3|3|100.00%|[]|

**Review by decster — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by nshangyiming** [comment]
run starrocks_admit_test

**Pull request merged**